internal/supervisor/supervisor.go

```diff
@@ -0,0 +1,111 @@
+package supervisor
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/mattermost/awat/internal/store"
+	"github.com/mattermost/awat/internal/translator"
+	"github.com/mattermost/awat/model"
+)
+
+// Supervisor is responsible for scheduling and launching Translations
+// in series
+type Supervisor struct {
+	logger  log.FieldLogger
+	store   *store.SQLStore
+	bucket  string
+	workdir string
+}
+
+// NewSupervisor returns a Supervisor prepared with the needed
+// metadata to operate
+func NewSupervisor(store *store.SQLStore, logger log.FieldLogger, bucket, workdir string) *Supervisor {
+	return &Supervisor{
+		store:   store,
+		logger:  logger.WithField("supervisor", model.NewID()),
+		bucket:  bucket,
+		workdir: workdir,
+	}
+}
+
+// Start runs the Supervisor's main routine on a new goroutine both
+// periodically and forever
+func (s *Supervisor) Start() {
+	s.logger.Info("Supervisor started")
+	go func() {
+		for {
+			s.supervise()
+			time.Sleep(60 * time.Second)
+		}
+	}()
+}
+
+// supervise queries the database for available Translations and
+// works through the batch returned serially
+func (s *Supervisor) supervise() {
+	work, err := s.store.GetTranslationsReadyToStart()
+	if err != nil {
+		s.logger.WithError(err).Error("Failed to query database for pending translations")
+		return
+	}
+
+	if len(work) > 0 {
+		s.logger.Debugf("Found %d requests pending to be translated", len(work))
+	}
+
+	for _, translation := range work {
```

Comment:
Correct me if I am wrong here but it seems to me that the following scenario is possible:
- We have 2 translations to work on and 2 instances running
- Instance A picks up translation 1 and starts working on it
- Instance B sees translation 1 as locked so it picks up translation 2
- Instance B finishes releasing lock.
- Instance A finishes and picks up translation 2 (again).

Reply:
Ah, I considered the scenario where B is still working on translation 2 (in which case A would fail to lock it) but overlooked the scenario where 2 is completed. 

I am going to change this so that instead of getting a list of work, each time it queries it will get back one Translation instead of a list of them. Basically it will use, the same pattern used in `GetAndClaimNextReadyImport`. 

I'd already been considering that but thought I had considered all possible race conditions here, so didn't think it was necessary to change my initial implementation. Great catch @Szymongib